python/src/opendp/extras/sklearn/linear_model/__init__.py

```diff
@@ -0,0 +1,125 @@
+'''
+This module requires extra installs: ``pip install 'opendp[scikit-learn]'``
+
+For convenience, all the functions of this module are also available from :py:mod:`opendp.prelude`.
+We suggest importing under the conventional name ``dp``:
+
+.. code:: python
+
+    >>> import opendp.prelude as dp
+
+The methods of this module will then be accessible at ``dp.sklearn.linear_model``.    
+
+If you're interested in the underlying algorithm, we've also
+`implemented Theil-Sen Regression as a demonstration of OpenDP plugins <../user-guide/plugins/theil-sen-regression.html>`_.
+'''
+
+from typing import Iterable
+from opendp.extras.sklearn.linear_model._make_private_theil_sen import (
+    make_private_theil_sen as _make_private_theil_sen,
+)  # noqa: F401
+from opendp._lib import import_optional_dependency
+from opendp.mod import assert_features
+
+
+class LinearRegression:
+    """
+    DP Linear Regression
+
+    The interface is parallel to that offered by sklearn's
+    `LinearRegression <https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html>`_.
+    The ``fit`` method returns an sklearn ``LinearRegression`` object.
+
+    :param fraction_bounds: Lower and upper bounds x_bounds to retain when ``fit`` is called.
+    """
+    def __init__(self, fraction_bounds: tuple[float, float] = (0.25, 0.75)):
+        self.fraction_bounds = fraction_bounds
+
+    def fit(
+        self,
+        X,
+        y,
+        x_bounds: Iterable[tuple[float, float]],
+        y_bounds: tuple[float, float],
+        scale: float,
+        runs: int = 1,
+        candidates_count: int = 100,
```

Comment:
All of these arguments too. Basically everything except the data. This way there will be 
1. enough arguments to be able to build the measurement without the data, for use in Context API
2. the fit method will more closely match the sklearn fit method